java-spiffe-core/src/main/java/spiffe/spiffeid/TrustDomain.java

```diff
@@ -0,0 +1,67 @@
+package spiffe.spiffeid;
+
+
+import lombok.NonNull;
+import lombok.Value;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static java.lang.String.format;
+
+/**
+ * A <code>TrustDomain</code> represents a normalized SPIFFE trust domain (e.g. domain.test).
+ */
+@Value
+public class TrustDomain {
+
+    String name;
+
+    private TrustDomain(String trustDomain) {
+        this.name = trustDomain;
+    }
+
+    /**
+     * Creates a trust domain.
+     *
+     * @param trustDomain a trust domain represented as a string, must not be blank.
+     * @return an instance of a {@link TrustDomain}
+     *
+     * @throws IllegalArgumentException if the given string is blank or cannot be parsed
+     */
+    public static TrustDomain of(@NonNull String trustDomain) {
+        if (StringUtils.isBlank(trustDomain)) {
+            throw new IllegalArgumentException("Trust Domain cannot be empty");
+        }
+        try {
+            val uri = new URI(normalize(trustDomain));
+            val host = getHost(uri);
```

Comment:
Seems that getHost() returns the URI path if the host is empty, so you could end up creating a TrustDomain of a path if trustDomain is something like "spiffe:///some/path" I believe. This shouldn't be allowed.